From 4db8815871797cd879c68b2aaa6ec29e3c6a20d0 Mon Sep 17 00:00:00 2001
From: dghart <dghart david@4Pane.co.uk>
Date: Thu, 7 Nov 2019 22:11:19 +0000
Subject: [PATCH] Linux compilation fix for recent pango versions

For some reason, pango-1.44 (needed by gtk+) has stopped looking for harfbuzz headers in /usr/include/harfbuzz.
This results in 'fatal error: hb.h: No such file or directory' when building for fedora 31. Cmake know about this,
see https://gitlab.kitware.com/cmake/cmake/issues/19531, and will eventually fix it. Meanwhile add
/usr/include/harfbuzz to CXXFLAGS. This fixes the problem without breaking earlier pango versions.
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb6df79c9..5f70148e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,9 @@ set( CL_COPY_WX_LIBS 0 )
 set( WITH_SFTP 1 )
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations") # Avoid very multiple warnings spam due to deprecated wx methods
+if( UNIX AND NOT APPLE)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -isystem /usr/include/harfbuzz") # Needed for fedora 31 so far, but will spread. See https://gitlab.kitware.com/cmake/cmake/issues/19531
+endif()
 
 if( UNIX AND NOT APPLE)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--disable-new-dtags")
-- 
2.23.0

